src/torrent.cpp

```diff
@@ -2759,16 +2759,14 @@ bool is_downloading_state(int const st)
 			// amount to exceed the total size of the torrent which upsets some trackers
 			if (is_seed())
 			{
-				for (peer_iterator i = m_connections.begin();
-					i != m_connections.end(); ++i)
+				for (auto c : m_connections)
 				{
 					TORRENT_INCREMENT(m_iterating_connections);
-					peer_connection* p = *i;
-					boost::optional<piece_block_progress> pbp = p->downloading_piece_progress();
-					if (pbp && pbp->bytes_downloaded > 0)
+					auto const pbp = c->downloading_piece_progress();
+					if (pbp.bytes_downloaded > 0)
```

Comment:
Don’t you still need to check that the optional is set?

Reply:
`downloading_piece_progress` does not return an optional since last year in `master`